spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/sql/AppEngineJdbcInfoProviderAutoConfiguration.java

```diff
@@ -0,0 +1,60 @@
+/*
+ *  Copyright 2017 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.autoconfigure.sql;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.cloud.gcp.autoconfigure.core.AppEngineCondition;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Provides the {@link CloudSqlJdbcInfoProvider} for apps running on Google App Engine.
+ *
+ * @author João André Martins
+ */
+@Configuration
+@ConditionalOnClass({com.google.cloud.sql.mysql.SocketFactory.class, GcpCloudSqlProperties.class})
```

Comment:
What about PostgreSQL support on App Engine standard?

Reply:
There's nothing specifically about PostgreSQL in GAE's documentation.
If we want to do anything there, then I'd do it in another PR.

Reply:
There is [documentation for Java 8](https://cloud.google.com/appengine/docs/standard/java/cloud-sql/using-cloud-sql-postgres), but I guess it's no different than connecting from App Engine flexible.

Does this mean that CloudSQL-MySQL on App Engine standard Java 8, could work without this special configuration? @saturnism

Reply:
AFAICT MySQL on App Engine has that special `<use-google-connector-j>true</use-google-connector-j>` element, which I think is us telling App Engine that the special driver should be used.

https://cloud.google.com/appengine/docs/standard/java/cloud-sql/using-cloud-sql-mysql